addons/test_mail/tests/test_mail_composer.py

```diff
@@ -1730,109 +1843,149 @@ def test_mail_composer_wtpl_complete(self):
+                # launch composer in mass mode
+                composer_form = Form(self.env['mail.compose.message'].with_context(ctx))
+                composer = composer_form.save()
 
-            # simulate cron queue at right time for sending
-            with freeze_time(self.reference_now + timedelta(days=2)):
-                self.env['mail.mail'].sudo().process_email_queue()
+                # ensure some parameters used afterwards
+                author = self.env.user.partner_id
+                self.assertEqual(composer.author_id, author,
+                                 'Author cannot be synchronized with a raw email_from')
+                self.assertEqual(composer.email_from, self.template.email_from)
 
-        self.assertEqual(len(self._mails), 10, 'Should have sent 5 emails per record')
+                with self.mock_mail_gateway(mail_unlink_sent=False), \
+                     freeze_time(self.reference_now):
+                    composer._action_send_mail()
 
-        # hack to use assertEmails: filtering on from/to only is not sufficient to distinguish emails
-        _mails_records = [
-            [mail for mail in self._mails if '%s-%s' % (record.id, record._name) in mail['message_id']]
-            for record in self.test_records
-        ]
+                    new_partners = self.env['res.partner'].search([
+                        ('email', 'in', [email_to_1, email_to_2, email_to_3, email_cc_1])
+                    ])
+                    self.assertEqual(len(new_partners), 3)
+
+                    if not scheduled_date:
+                        # global outgoing: emails sent
+                        self.assertEqual(len(self._new_mails), 2, 'Should have created 1 mail.mail per record')
+                        self.assertEqual(len(self._mails), 10, 'Should have sent emails')
+                        self.assertEqual(self._new_mails.mapped('scheduled_date'),
+                                         [False] * 2)
+                    else:
+                        # global outgoing: emails not sent due to scheduled_date
+                        self.assertEqual(len(self._new_mails), 2, 'Should have created 1 mail.mail per record')
+                        self.assertEqual(len(self._mails), 0, 'Should not send emails, scheduled in the future')
+                        self.assertEqual(self._new_mails.mapped('scheduled_date'),
+                                         [self.reference_now + timedelta(days=2)] * 2)
+
+                        # simulate cron queue at right time for sending
+                        with freeze_time(self.reference_now + timedelta(days=2)):
+                            self.env['mail.mail'].sudo().process_email_queue()
 
-        for record, _mails in zip(self.test_records, _mails_records):
-            # message copy is kept
-            message = record.message_ids[0]
+                self.assertEqual(len(self._mails), 10, 'Should have sent 5 emails per record')
```

Comment:
Was already tested for the "not scheduled_date" case so could be moved inside the else part ?